tap_playvox/sync.py

```diff
@@ -159,10 +186,19 @@ def sync_endpoint(client,
                         
                         if len(date_records)==parsed_date_records:
                             parse = False
-                            
+
+        # Subtract 14 days from the current date and time for Schedule Metrics
+        startDate = datetime.utcnow() - timedelta(days=14)
+
+        # Set the time to midnight
+        startDate = startDate.replace(hour=0, minute=0, second=0, microsecond=0)
+                   
         # 1. set bookmark
-        singer.write_bookmark(state, stream_name, 'endDate', str(datetime.utcnow().isoformat()))
-                
+        if(stream_name == 'schedule_metrics'):
+            singer.write_bookmark(state, stream_name, 'endDate', str(startDate.isoformat()))
+        else:
+            singer.write_bookmark(state, stream_name, 'endDate', str(datetime.utcnow().isoformat()))
+            
```

Comment:
why are we writing endDate as a bookmark in the first place? this should be some timestamp field, but end date can be confusing. you can imagine that a tap runs and finishes on 2023-08-30, on the next start time is 2023-08-31

Reply:
I will update the code to use the DateTime field of the object.

Reply:
@vorpal56 - I checked Playvox API documentation. The data sent is always from start_date specified in the param of the URL to current_date_time, unless end_date is specified. Given how this API is designed, storing current date_time for bookmarks should ingest data as expected without gaps.